nautobot_circuit_maintenance/handle_notifications/handler.py

```diff
@@ -235,22 +219,75 @@ def process_raw_notification(  # pylint: disable=too-many-branches
 
     logger.log_success(raw_entry, message="Raw notification created.")
 
-    for parsed_notification in parsed_notifications:
+    return raw_entry
+
+
+def process_raw_notification(logger: Job, notification: MaintenanceNotification) -> Optional[uuid.UUID]:
+    """Processes a raw notification (maybe containing multiple parsed notifications).
+
+    It creates a RawNotification and if it could be parsed, create the corresponding ParsedNotification and the
+    related objects. Finally returns the the UUID of the RawNotification modified.
+    """
+    provider = Provider.objects.filter(slug=notification.provider_type).last()
+    if not provider:
+        logger.log_warning(
+            message=(
+                f"Raw notification not created because is referencing to a provider not existent: {notification.provider_type}"
+            )
+        )
+        return None
+
+    raw_entry = create_raw_notification(logger, notification, provider)
+    if not raw_entry:
+        return None
+
+    parser_maintenances = get_maintenances_from_notification(logger, notification, provider)
+    if not parser_maintenances:
+        return raw_entry.id
+
+    for parser_maintenance in parser_maintenances:
         try:
-            process_parsed_notification(logger, parsed_notification, raw_entry, provider)
-            # Update raw notification as properly parsed
+            circuit_maintenance_entry = create_or_update_circuit_maintenance(
+                logger, parser_maintenance, raw_entry, provider
+            )
+            # Update raw notification as properly parsed and with the stamp time
             raw_entry.parsed = True
+            raw_entry.date = datetime.datetime.fromtimestamp(parser_maintenance.stamp)
             raw_entry.save()
```

Comment:
Can we do these updates atomically as part of `create_or_update_circuit_maintenance` instead of as a separate step? I have an extension plugin that's listening for the `post_save` signal for `CircuitMaintenance` and I think with these changes it'll be triggered twice instead of once.

Reply:
because this is for `RawNotification` I intentionally put outside of the `CircuitMaintenance` manipulation. I guess this should affect the signal, no?

Reply:
Ah, my mistake! Fine with me to update the raw entry separately.